### PR TITLE
feat: initialize github workflow and action

### PR DESCRIPTION
**Pull request by mikevespi**
Initialize github actions / workflows

### CODE REVIEW

File: .github/actions/dev-env-setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        echo "PERL_MB_OPT=--install_base '/home/runner/perl5'" >> $GITHUB_ENV
+        echo "PERL_MM_OPT=INSTALL_BASE=/home/runner/perl5" >> $GITHUB_ENV
+        echo "/home/runner/perl5/bin" >> $GITHUB_PATH
+        echo "CURRENT_PERL_VERSION=$(perl -e 'print "$^V\n"')" >> $GITHUB_ENV
+    - uses: actions/cache@v3
+      id: asdf-cache
+      with:
+        path: |
+          ~/.asdf
+        key: ${{ runner.os }}-asdf-cache-${{ hashFiles('.tool-versions') }}-v2
```
Comment (matthieu-foucault) [inline .github/actions/dev-env-setup/action.yml]: you don't need those `v2`, they were added to invalidate some bad caches. Also we can delete caches with the github API now, so this shouldn't be needed ever again

File: .github/workflows/test-code.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+jobs:
+  install-dev-tools:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: dev env setup
+        uses: ./.github/actions/dev-env-setup
+      - run: make install_dev_tools
+      - run: make stop_pg # cleanly stop the database, otherwise it will complain in the subsequent jobs
+      - run: yarn install --frozen-lockfile
```
Comment (matthieu-foucault) [inline .github/workflows/test-code.yml]:
```suggestion
      - run: yarn install --frozen-lockfile --check-files
```

File: .github/workflows/test-code.yml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        uses: actions/checkout@v3
+      - name: Run Trivy vulnerability scanner in repo mode
+        uses: aquasecurity/trivy-action@master
+        with:
+          scan-type: fs
+          format: sarif
+          output: trivy-results.sarif
+          exit-code: "0"
+          ignore-unfixed: false
+          severity: UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL
```
Comment (matthieu-foucault) [inline .github/workflows/test-code.yml:43]: you might want to increase the trivy timeout to 10 min, see the latest CCBC updates

File: .github/workflows/test-code.yml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+          if test -f ".git/COMMIT_EDITMSG"; then
+              echo "COMMIT_EDITMSG EXISTS, skipping"
+          else
+              touch .git/COMMIT_EDITMSG
+          fi
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ./.pre-commit-cache
+          key: pre-commit-${{ env.PY }}-${{ hashFiles('.pre-commit-config.yaml') }}-v3
```
Comment (matthieu-foucault) [inline .github/workflows/test-code.yml]:
```suggestion
          key: pre-commit-${{ env.PY }}-${{ hashFiles('.pre-commit-config.yaml') }}
```

File: app/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+module.exports = {
+  extends: ["airbnb-typescript", "next", "prettier"],
```
Comment (matthieu-foucault) [inline app/.eslintrc.js:2]: add `airbnb` too?

Comment (mikevespi) [reply app/.eslintrc.js:2]: I ended up removing airbnb (not their typescript though). Some of their rules were intense, and throwing errors in the out of the box next code. I'll try adding it again if / when we start working on a frontend for this and give them a proper read though.

**Pull request merged**